src/pages/settings/Payments/TransferBalancePage.js

```diff
@@ -1,27 +1,251 @@
+import compose from '../../../libs/compose';
 import KeyboardAvoidingView from '../../../components/KeyboardAvoidingView';
+import * as Expensicons from '../../../components/Icon/Expensicons';
+import MenuItem from '../../../components/MenuItem';
+import CONST from '../../../CONST';
+import variables from '../../../styles/variables';
+import Text from '../../../components/Text';
+import Button from '../../../components/Button';
+import FixedFooter from '../../../components/FixedFooter';
+import CurrentWalletBalance from '../../../components/CurrentWalletBalance';
+import walletTransferPropTypes from './walletTransferPropTypes';
+import * as PaymentMethods from '../../../libs/actions/PaymentMethods';
+import PaymentUtils from '../../../libs/PaymentUtils';
+import userWalletPropTypes from '../../EnablePayments/userWalletPropTypes';
 
 const propTypes = {
+    /** User's wallet information */
+    userWallet: userWalletPropTypes.userWallet,
+
+    /** Array of bank account objects */
+    bankAccountList: PropTypes.arrayOf(PropTypes.shape({
+        /** The name of the institution (bank of america, etc) */
+        addressName: PropTypes.string,
+
+        /** The masked bank account number */
+        accountNumber: PropTypes.string,
+
+        /** The bankAccountID in the bankAccounts db */
+        bankAccountID: PropTypes.number,
+
+        /** The bank account type */
+        type: PropTypes.string,
+    })),
+
+    /** Array of card objects */
+    cardList: PropTypes.arrayOf(PropTypes.shape({
+        /** The name of the institution (bank of america, etc) */
+        cardName: PropTypes.string,
+
+        /** The masked credit card number */
+        cardNumber: PropTypes.string,
+
+        /** The ID of the card in the cards DB */
+        cardID: PropTypes.number,
+    })),
+
+    /** Wallet balance transfer props */
+    walletTransfer: walletTransferPropTypes,
+
     ...withLocalizePropTypes,
 };
 
-const TransferBalancePage = props => (
-    <ScreenWrapper>
-        <KeyboardAvoidingView>
-            <HeaderWithCloseButton
-                title={props.translate('common.transferBalance')}
-                shouldShowBackButton
-                onBackButtonPress={() => Navigation.goBack()}
-                onCloseButtonPress={() => Navigation.dismissModal(true)}
-            />
-        </KeyboardAvoidingView>
-    </ScreenWrapper>
-);
+const defaultProps = {
+    userWallet: {},
+    bankAccountList: [],
+    cardList: [],
+    walletTransfer: {},
+};
+
+class TransferBalancePage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.paymentTypes = [
+            {
+                key: CONST.WALLET.TRANSFER_METHOD_TYPE.INSTANT,
+                title: this.props.translate('transferAmountPage.instant'),
+                description: this.props.translate('transferAmountPage.instantSummary', {
+                    amount: this.props.numberFormat(
+                        0.25,
```

Comment:
We actually want this to be 1.5% of the total transfer, with a $0.25 minimum

Reply:
Ok. So I will add another method to calculate the real fees. I guess the Fee on the UI will also change.

Reply:
Section 5 in the original issue has said that since the beginning I believe 😉 https://github.com/Expensify/App/issues/3922